pkg/gadgets/trace/dns/tracer/bpf/dns.c

```diff
@@ -104,19 +105,26 @@ int ig_trace_dns(struct __sk_buff *skb)
 	}
 
 	__u32 len = i < MAX_DNS_NAME ? i : MAX_DNS_NAME;
+	if  (len == 0)
+		return 0;
 
 	struct event_t event = {0,};
-	if (len > 0)
-		bpf_skb_load_bytes(skb, DNS_OFF + sizeof(struct dnshdr), event.name, len);
+	event.af = AF_INET;
+	event.daddr_v4 = load_word(skb, ETH_HLEN + offsetof(struct iphdr, daddr));
+	event.saddr_v4 = load_word(skb, ETH_HLEN + offsetof(struct iphdr, saddr));
+	// load_word converts from network to host endianness. Convert back to
+	// network endianness.
```

Comment:
We need to have it in network byte order because we are using `inet_ntop` to transform it into text form. right?